### PR TITLE
fix(smde): Fix SMDE and remove usage of IFluidHTMLView

### PR DESCRIPTION
**Pull request by ChumpChief**
This change does the following:

- Lock the version of codemirror used by smde to one which is compatible with its usage of the addons.  This gets SMDE working again, it's currently broken in `main`
- Splits SmdeView from SmdeDataObject - didn't try for a super-clean interface, but it's enough to extract the render function from the data object.
- Wraps the view with a React wrapper so we can avoid using IFluidHTMLView (same technique as in #13969)
- Drops usage of IFluidHTMLView - this was the last usage in the repo outside of the view-interfaces/view-adapters directories

### CODE REVIEW

**Comment by tylerbutler** [comment]
@ChumpChief Glad you were able to figure out the code mirror issue. Any thoughts on using overrides to set the dep version rather than manual lockfile edit? I think it's a little more explicit and easier to trace. https://pnpm.io/package_json#pnpmoverrides

**Comment by ChumpChief** [comment]
packageExtensions seems to do what I want!  Tested that it automatically regenerated the lockfile back to the state I want too.

File: examples/data-objects/smde/src/smde.ts [hunk 1]
```diff
@@ -15,28 +15,20 @@ import {
...
-import { getTextAndMarkers, SharedString } from "@fluidframework/sequence";
-import { IFluidHTMLOptions, IFluidHTMLView } from "@fluidframework/view-interfaces";
-import SimpleMDE from "simplemde";
+import { SharedString } from "@fluidframework/sequence";
 
 // eslint-disable-next-line import/no-internal-modules, import/no-unassigned-import
 import "simplemde/dist/simplemde.min.css";
 
-export class Smde extends EventEmitter implements IFluidLoadable, IFluidRouter, IFluidHTMLView {
+export class SmdeDataObject extends EventEmitter implements IFluidLoadable, IFluidRouter {
```
Comment (Josmithr) [inline examples/data-objects/smde/src/smde.ts:29]: Docs? 🙂

File: examples/data-objects/smde/src/smdeView.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+		this.smde = smde;
+
+		const { parallelText } = getTextAndMarkers(this.smdeDataObject.text, "pg");
+		const text = parallelText.join("\n");
+		this.smde.value(text);
+
+		let localEdit = false;
+
+		this.smdeDataObject.text.on("sequenceDelta", (ev) => {
+			if (ev.isLocal) {
```
Comment (Josmithr) [inline examples/data-objects/smde/src/smdeView.tsx:64]: Nit: probably worth a brief comment as to why we want to early-out for local events.

File: examples/data-objects/smde/src/smdeView.tsx [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+				return;
+			}
+
+			localEdit = true;
+			for (const range of ev.ranges) {
+				const segment = range.segment;
+
+				if (range.operation === MergeTreeDeltaType.INSERT) {
+					if (TextSegment.is(segment)) {
+						// TODO need to count markers
```
Comment (Josmithr) [inline examples/data-objects/smde/src/smdeView.tsx:74]: Is this intended to be done separately? Do we have anything tracking it? What are the implications of this not being done?

File: examples/data-objects/smde/src/smdeView.tsx [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+	private setupEditor() {
+		const smde = new SimpleMDE({ element: this.textArea });
+		this.smde = smde;
+
+		const { parallelText } = getTextAndMarkers(this.smdeDataObject.text, "pg");
+		const text = parallelText.join("\n");
+		this.smde.value(text);
+
+		let localEdit = false;
```
Comment (Josmithr) [inline examples/data-objects/smde/src/smdeView.tsx:61]: It wasn't immediately clear to me what this value is tracking. Some brief docs here might be helpful too.

File: examples/data-objects/smde/src/smdeView.tsx [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+						[reservedTileLabelsKey]: ["pg"],
+					});
+					from++;
+				}
+			});
+		});
+	}
+}
+
+export interface ISmdeReactViewProps {
```
Comment (Josmithr) [inline examples/data-objects/smde/src/smdeView.tsx:138]: Docs for the exported members? 😁

File: examples/data-objects/smde/src/index.ts [hunk 6]
```diff
@@ -3,18 +3,48 @@
...
+import React from "react";
+
+import { SmdeDataObject, SmdeFactory } from "./smde";
+import { SmdeReactView } from "./smdeView";
 
 const defaultComponentId = "default";
 
 const smde = new SmdeFactory();
 
+const viewRequestHandler = async (request: RequestParser, runtime: IContainerRuntime) => {
```
Comment (Josmithr) [inline examples/data-objects/smde/src/index.ts:28]: I'm not familiar with our view pattern. Might be nice to add some brief docs to this for folks who are unfamiliar. Also, probably worth specifying the return type.

**Review by Josmithr — APPROVED** [review]
Changes look good to me, but there are a number of areas here I am unfamiliar with. I left some suggestions where I think extra docs would be beneficial to fellow newbies to our view patterns.

Also, probably not in scope for this PR, but it might be beneficial to upgrade the eslint config in this package to at least `recommended` and clean up the violations. In particular, a lot of function-like members are missing return types.

Might also be nice to use the Conventional Commits format for the PR title too, since I believe we are moving in that direction.

File: examples/data-objects/smde/src/smdeView.tsx (hunk 5 above)
Comment (ChumpChief) [reply examples/data-objects/smde/src/smdeView.tsx:135]: Good callout - helped me realize that I had left a duplicate and that an entire file/dependency was unused :-P

File: examples/data-objects/smde/src/index.ts (hunk 6 above)
Comment (ChumpChief) [reply examples/data-objects/smde/src/index.ts:28]: I'm going to leave this one - container-views is the better/dedicated example of what this is doing for the average customer, but I'd personally even recommend against that.  Preferred path forward would be to get rid of the webpack-fluid-loader and then this pattern goes away.

File: examples/data-objects/smde/src/smdeView.tsx (hunk 4 above)
Comment (ChumpChief) [reply examples/data-objects/smde/src/smdeView.tsx:54]: I honestly don't know either :-P. Going to leave this as it was preexisting.

File: examples/data-objects/smde/src/smdeView.tsx (hunk 3 above)
Comment (ChumpChief) [reply examples/data-objects/smde/src/smdeView.tsx:68]: Not sure - I'm going to leave this as it was preexisting.

File: examples/data-objects/smde/src/smdeView.tsx (hunk 2 above)
Comment (ChumpChief) [reply examples/data-objects/smde/src/smdeView.tsx:58]: Done in latest

**Comment by ChumpChief** [comment]
I'll leave the shift to eslint recommended, but at least removed the strict-boolean-expressions exception :)

**Pull request merged**